lib/src/over_react_test/common_component_util.dart

```diff
@@ -147,145 +169,158 @@ void expectCleanTestSurfaceAtEnd() {
-//      expect(actualProps[defaultTestIdKey], contains(testId),
-//          reason: '$defaultTestIdKey was not forwarded or was forwarded and then overridden.');
-//
-//      var ambiguousProps = {};
-//
-//      Set propKeysThatShouldNotGetForwarded = propsThatShouldNotGetForwarded.keys.toSet();
-//      // Don't test any keys specified by skippedPropKeys.
-//      propKeysThatShouldNotGetForwarded.removeAll(skippedPropKeys);
-//
-//      Set unexpectedKeys = actualProps.keys.toSet().intersection(propKeysThatShouldNotGetForwarded);
-//
-//      /// Test for prop keys that both are forwarded and exist on the forwarding target's default props.
-//      if (isDartComponent(forwardingTarget)) {
-//        // ignore: avoid_as
-//        var forwardingTargetDefaults = ((forwardingTarget as ReactElement).type as ReactClass).dartDefaultProps;
-//
-//        var commonForwardedAndDefaults = propKeysThatShouldNotGetForwarded
-//            .intersection(forwardingTargetDefaults.keys.toSet());
-//
-//        /// Don't count these as unexpected keys in later assertions; we'll verify them within this block.
-//        unexpectedKeys.removeAll(commonForwardedAndDefaults);
-//
-//        commonForwardedAndDefaults.forEach((propKey) {
-//          var defaultTargetValue = forwardingTargetDefaults[propKey];
-//          var potentiallyForwardedValue = propsThatShouldNotGetForwarded[propKey];
-//
-//          if (defaultTargetValue != potentiallyForwardedValue) {
-//            /// If the potentially forwarded value and the default are different,
-//            /// we can tell whether it was forwarded.
-//            expect(actualProps, isNot(containsPair(propKey, potentiallyForwardedValue)),
-//                reason: 'The `$propKey` prop was forwarded when it should not have been');
-//          } else {
-//            /// ...otherwise, we can't be certain that the value isn't being forwarded.
-//            ambiguousProps[propKey] = defaultTargetValue;
-//          }
-//        });
-//      }
-//
-//      expect(unexpectedKeys, isEmpty, reason: 'Should filter out all consumed props');
-//
-//      if (ambiguousProps.isNotEmpty) {
-//        fail(unindent(
-//            '''
-//            Encountered ambiguous forwarded props; some unconsumed props coincide with defaults on the forwarding target, and cannot be automatically tested.
-//
-//            Try either:
-//              - specifying `nonDefaultForwardingTestProps` as a Map with valid prop values that are different than the following: $ambiguousProps
-//              - specifying `skippedPropKeys` with the following prop keys and testing their forwarding manually: ${ambiguousProps.keys.toList()}
-//            '''
-//        ));
-//      }
-//    }
-//  });
-//}
+/// todo make this public again if there's a need to expose it, once the API has stabilized
+void _testPropForwarding(BuilderOnlyUiFactory factory, dynamic childrenFactory(), {
```

Comment:
This implementation is exactly the same, minus 
- some Dart 2 syntax updates
- updates to use `meta`
- changes to not use `dartDefaultProps` on Component2, since it doesn't apply